fastify.js

```diff
@@ -348,6 +348,12 @@ function fastify (options) {
   }
 
   Object.defineProperties(fastify, {
+    listeningOrigin: {
+      get () {
+        const ipv4Address = this.addresses().find(address => address.family === 'IPv4')
+        return `${this[kOptions].https ? 'https' : 'http'}://${ipv4Address.address}:${ipv4Address.port}`
```

Comment:
The implementation is ignoring the fact that Fastify listens to both lookup IP address families when `localhost` or `::` (without `ipv6only`) are passed when executing `fastify.listen`.

Not to mention also the possible Unix domain socket path that can be used instead.

Maybe we should return the _last_ item of the array returned by `fastify.addresses` method?
This as the last item contains the information about the primary server in case the scenario described previously takes place. Reference: [`Fastify#addresses`](https://github.com/fastify/fastify/blob/fc9e7c02eca13f26fbd5db37af1052e332c741c3/fastify.js#L326)

Reply:
Sure, I can use the last item of the array returned by `fastify.addresses`.

However, to be honest, my preferred solution would be a simpler one.
Just assign the value to `listeningOrigin` inside the `listen` function (in `lib/server.js`), something like this:

<img width="880" alt="immagine" src="https://user-images.githubusercontent.com/4263614/232830941-360d2636-5127-4671-909c-d02d7eed9dca.png">

What do you think?
Of course, in this case `listeningOrigin` would  be mutable

Reply:
That will work for the combination of `host:port` but for Unix domain sockets it will show `http://undefined:undefined`.

The `listeningOrigin` should consider all the scenarios `fastify` supports 🙂

Reply:
Ok, let's use the last item of the array returned by `fastify.addresses`, thanks!